packages/hub/routes/email-card-drop-requests.ts

```diff
@@ -85,12 +96,46 @@ export default class EmailCardDropRequestsRoute {
 
     ctx.type = 'application/vnd.api+json';
 
+    let prepaidCardMarketV2 = await this.cardpay.getSDK('PrepaidCardMarketV2', this.web3.getInstance());
+
+    if (await prepaidCardMarketV2.isPaused()) {
+      return respondWith503(ctx, 'The prepaid card market contract is paused');
+    }
+
+    let quantityAvailable = await prepaidCardMarketV2.getQuantity(cardDropSku);
+
+    let quantityIsBelowNotificationThreshold = quantityAvailable < notifyWhenQuantityBelow;
+    let notificationHasNotBeenSentRecently =
+      !lastNotifiedOnQuantity || lastNotifiedOnQuantity < Date.now() - millisToWaitBeforeNotifying;
+
+    log.trace(
+      `${cardDropSku} has ${quantityAvailable} available, notification threshold is ${notifyWhenQuantityBelow}`
+    );
+
+    if (quantityIsBelowNotificationThreshold && notificationHasNotBeenSentRecently) {
+      Sentry.captureException(
+        new Error(
+          `Prepaid card quantity (${quantityAvailable}) is below cardDrop.email.notifyWhenQuantityBelow threshold of ${notifyWhenQuantityBelow}`
+        ),
+        {
+          tags: {
+            action: 'drop-card',
+            alert: 'web-team',
+          },
+        }
+      );
+
+      lastNotifiedOnQuantity = Date.now();
+    }
+
+    let activeReservations = await this.emailCardDropRequestQueries.activeReservations();
+
+    if (quantityAvailable <= activeReservations) {
+      return respondWith503(ctx, 'There are no prepaid cards available');
```

Comment:
This should probably also trigger a Sentry alert, since this is the "out of prepaid card" situation we're trying to prevent/respond to. It is possible to be a false positive, but I think we would want to know anyway since this is affecting user experience.

Reply:
Now that I‘ve included reservations in the threshold calculation, I think this would be redundant…??